tests/test_oracle.py

```diff
@@ -0,0 +1,214 @@
+                "".join([str(i)[-1] for i in range(1_024)]),
+            ]
+        },
+        dtype="string",
+    )
+    wr.oracle.to_sql(
+        df=df,
+        con=oracle_con,
+        table=table,
+        schema="TEST",
+        mode="overwrite",
+        index=False,
+        dtype={"col": "VARCHAR(1024)"},
+    )
+    df2 = wr.oracle.read_sql_query(sql=f'SELECT * FROM "TEST"."{table}"', con=oracle_con)
+    assert df.equals(df2)
+
+
+def test_null(oracle_table, oracle_con):
+    table = oracle_table
+    df = pd.DataFrame({"id": [1, 2, 3], "nothing": [None, None, None]})
+    wr.oracle.to_sql(
+        df=df,
+        con=oracle_con,
+        table=table,
+        schema="TEST",
+        mode="overwrite",
+        index=False,
+        dtype={"nothing": "INTEGER"},
+    )
+    wr.oracle.to_sql(
+        df=df,
+        con=oracle_con,
+        table=table,
+        schema="TEST",
+        mode="append",
+        index=False,
+    )
+
+    df2 = wr.oracle.read_sql_table(table=table, schema="TEST", con=oracle_con)
+    df["id"] = df["id"].astype("Int64")
+    assert pd.concat(objs=[df, df], ignore_index=True).equals(df2)
+
+
+def test_decimal_cast(oracle_table, oracle_con):
+    table = oracle_table
+    df = pd.DataFrame(
+        {
+            "col0": [Decimal((0, (1, 9, 9), -2)), None, Decimal((0, (1, 9, 0), -2))],
+            "col1": [Decimal((0, (1, 9, 9), -2)), None, Decimal((0, (1, 9, 0), -2))],
+            "col2": [Decimal((0, (1, 9, 9), -2)), None, Decimal((0, (1, 9, 0), -2))],
+        }
+    )
+    wr.oracle.to_sql(df, oracle_con, table, "TEST")
+    df2 = wr.oracle.read_sql_table(
+        schema="TEST", table=table, con=oracle_con, dtype={"col0": "float32", "col1": "float64", "col2": "Int64"}
+    )
+    assert df2.dtypes.to_list() == ["float32", "float64", "Int64"]
+    assert 3.88 <= df2.col0.sum() <= 3.89
+    assert 3.88 <= df2.col1.sum() <= 3.89
+    assert df2.col2.sum() == 2
+
+
+def test_read_retry(oracle_con):
+    try:
+        wr.oracle.read_sql_query("ERROR", oracle_con)
+    except:  # noqa
+        pass
+    df = wr.oracle.read_sql_query("SELECT 1 FROM DUAL", oracle_con)
+    assert df.shape == (1, 1)
+
+
+def test_table_name(oracle_con):
+    df = pd.DataFrame({"col0": [1]})
+    wr.oracle.to_sql(df, oracle_con, "Test Name", "TEST", mode="overwrite")
+    df = wr.oracle.read_sql_table(schema="TEST", con=oracle_con, table="Test Name")
+    assert df.shape == (1, 1)
+    with oracle_con.cursor() as cursor:
+        cursor.execute('DROP TABLE "Test Name"')
+    oracle_con.commit()
+
+
+@pytest.mark.parametrize("dbname", [None, "ORCL"])
+def test_connect_secret_manager(dbname):
+    try:
+        con = wr.oracle.connect(secret_id="aws-data-wrangler/oracle", dbname=dbname)
+        df = wr.oracle.read_sql_query("SELECT 1 FROM DUAL", con=con)
+        assert df.shape == (1, 1)
+    except boto3.client("secretsmanager").exceptions.ResourceNotFoundException:
```

Comment:
I think this test might be skewing the coverage from Abdel's comment above... are we sure this is still needed?